alfresco-actuators/src/main/java/eu/xenit/actuators/services/AlfrescoInfoService.java

```diff
@@ -0,0 +1,192 @@
+    private ServiceRegistry serviceRegistry;
+    @Resource(name="DescriptorService")
+    private DescriptorService descriptorService;
+    @Autowired
+    private ModuleService moduleService;
+    @Autowired(required=false)
+    private LicenseService licenseService;
+    @Autowired
+    @Qualifier("global-properties")
+    private Properties globalProperties;
+    @Autowired
+    private ApplicationContext appContext;
+    @Autowired
+    private RepoAdminService repoAdminService;
+    @Autowired
+    private AuditService auditService;
+    @Autowired
+    private ThumbnailService thumbnailService;
+    @Autowired
+    private RetryingTransactionHelper retryingTransactionHelper;
+
+
+    AlfrescoInfo getAlfrescoInfo() {
+
+        ManifestInfo manifestInfo = ManifestInfo.getInstance();
+        String id = null;
+        String version = null;
+        String edition = null;
+        LicenseInfo license = null;
+        // descriptorService not available in the DE version for alfresco 52 and alfresco 60
+        if(descriptorService!=null) {
+            Descriptor serverDescriptor = descriptorService.getServerDescriptor();
+            Descriptor repositoryDescriptor = descriptorService.getCurrentRepositoryDescriptor();
+            id = repositoryDescriptor.getId();
+            version = serverDescriptor.getVersion();
+            edition = serverDescriptor.getEdition();
+            license = retrieveLicenseInfo();
+        }
+
+        return new AlfrescoInfo()
+                .id(id)
+                .version(version)
+                .warManifest(manifestInfo.getManifestProperties())
+                .edition(edition)
+                .license(license)
+                .modules(this.retrieveAlfrescoModules())
+                .globalProperties(this.retrievePropertiesFiltered())
+                .status(this.retrieveStatusInfo());
+    }
+
+
+    String getVersion() {
+        Descriptor serverDescriptor = descriptorService.getServerDescriptor();
+        return serverDescriptor.getVersion();
+    }
+
+    String getEdition() {
+        Descriptor serverDescriptor = descriptorService.getServerDescriptor();
+        return serverDescriptor.getEdition();
+    }
+
+    Map<String, String> getManifestProperties() {
+        return ManifestInfo.getInstance().getManifestProperties();
+    }
+
+    private Map<String, String> retrievePropertiesFiltered() {
+        final String PROP_STARTS_WITH = "prefix.properties.filtered";
+
+        String propsShouldStartWith = globalProperties.getProperty(PROP_STARTS_WITH);
+        if (propsShouldStartWith == null) {
+            logger.error("Property '{}' is not set, you'll see no properties...", PROP_STARTS_WITH);
+            return null;
+        }
+
+        Map<String, String> map = new TreeMap<>();
+
+        for (Map.Entry<Object,Object> entry : globalProperties.entrySet()) {
+            String key = entry.getKey().toString();
+            if (key.startsWith(propsShouldStartWith)
```

Comment:
Isn't this functionality a bit dangerous? Setting your `prefix.properties.filtered` global property to the empty string will expose all properties.

Reply:
It's true that it will expose all properties. I don't see it as a problem.
Will ask for a second opinion.

Reply:
I would propose to sanitize out sensitive values.

Any key that contains `password`, `secret` or `key` -> replace the value with `*****`

Spring Boot uses a similar (configurable) sanitization strategy:
* Documentation: https://docs.spring.io/spring-boot/docs/current/reference/html/howto.html#howto-sanitize-sensitive-values
* Implementation/usage reference: https://github.com/spring-projects/spring-boot/blob/master/spring-boot-project/spring-boot-actuator/src/main/java/org/springframework/boot/actuate/context/properties/ConfigurationPropertiesReportEndpoint.java#L214

Reply:
Can be in a follow up PR, but it is important imo

Reply:
@tgeens If it is ok for you, I will remove this functionality for now. If it will be needed in the future for e.g. a particular http check, it is easy to add.

Reply:
ok - this is not needed for health checks anyway

Reply:
In the current system there are 2 checks implemented, which I was thinking to re-implement using this endpoint:

-basic health check
-read only check 

But, I realized there might be a security issue anyway: is it acceptable to expose information about license in an unauthenticated endpoint?

Reply:
Since it is not acceptable to expose any confidential information in the output of this script, this script just returns a:

{"status":"UP"} or {"status":"DOWN"}